che-incubator/che-code
# fix: extension secrets should not be lost when workspace restart

### What does this PR do?
Allows to keep encrypted extension secrets in the browser local storage.

![Screenshot from 2024-05-15 09-53-39](https://github.com/che-incubator/che-code/assets/1655894/ea396d7e-23cd-4e1e-8da1-7b2c4646563c)

Launcher output
```
# Injecting server public key to che-code...
  > foud key file /etc/ssh/default-1611150837256.pub
  > out/vs/code/browser/workbench/workbench.js has been updated
```

### What issues does this PR fix?

https://github.com/eclipse-che/che/issues/22837

### How to test this PR?

1. Create a workspace with
https://github.com/vitaliy-guliy/vscode-test-extension/tree/persisting-secrets?che-editor=https://gist.githubusercontent.com/vitaliy-guliy/4fedeef3dbf8e902af94240780c442e6/raw/f08e91485b8529c6761b73f375fccde8d62ae94f/editor.yaml&new

2. Build test extension from sources or download the binary from https://github.com/vitaliy-guliy/vscode-test-extension/releases/tag/0.0.2

3. Go to extensions view, install the extension from .vsix file

4. Run `Secrets: Add secret` command and add the secret entering any value

5. Delete the workspace and create it again with the link above, or just restart the workspace (chose what do you prefer). If you recreated workspace, it needs to install test extension again.

6. Run `Secrets: Get Secret` command. You should get the value entered before.

You can create any workspace using `che-code` image provided by this pull request and experiment with any extension you want. For instance, it is possible to install sonarlint extension, open extension settings, configure it and then restart the workspace. All the settings should be saved.

### Does this PR contain changes that override default upstream Code-OSS behavior?
